sessionize/app/src/main/java/co/touchlab/sessionize/MainActivity.kt

```diff
@@ -125,4 +133,16 @@ class MainActivity : AppCompatActivity(), NavigationHost, SnackHost {
 
         transaction.commit()
     }
+
+    private fun showFeedbackDialog(session: MySessions){
+        val feedbackDialog = FeedbackDialog()
+        feedbackDialog.showNow(supportFragmentManager, "FeedbackDialog")
+        feedbackDialog.setSessionInfo(session.id.toInt(),session.title)
+        feedbackDialog.setFeedbackDialogInterface(feedbackDialogInterface = object : FeedbackDialogInterface {
+            override fun finishedFeedback(sessionId: String, rating: Int, comment: String) {
+                AppContext.dbHelper.updateFeedback(rating.toLong(),comment,sessionId)
+                cancelLocalNotification(sessionId.toInt(), NotificationFeedbackTag)
```

Comment:
The feedback logic could probably be pushed into it's own "veiw helper" class, and the Feedback dialog would just get session id and manage it's own logic. In this case, 'updateFeedback' needs to be in a background thread, and sessionId.toInt() would fail. Better to centralize and test that stuff.

Reply:
I've centralized this into the FeedbackManager class.